tests/v2/integration/catalogv2/cluster_repo_test.go

```diff
@@ -280,14 +299,21 @@ func Start429Registry(t assert.TestingT, rateLimitedHeader bool) (*url.URL, erro
 			}
 
 		}
-	}))
+	})
 
-	u, err := url.Parse(ts.URL)
+	// Create a new test server
+	ts := httptest.NewUnstartedServer(customHandler)
+
+	ip := getOutboundIP()
+	// Bind the server to a specific IP address (your local machine's IP)
+	listener, err := net.Listen("tcp", fmt.Sprintf("%s:8080", ip.String()))
 	if err != nil {
-		return nil, err
+		log.Fatalf("Failed to bind to local IP: %v", err)
```

Comment:
My mistake can you change this to  return nil, err instead of a fatal?

Reply:
done!